src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostOptions.cs

```diff
@@ -17,19 +17,17 @@ namespace Microsoft.VisualStudio.LanguageServices.Remote
 {
     internal static class RemoteHostOptions
     {
-        // Update primary workspace on OOP every 4 seconds if VS is not running any global operation 
-        // such as build, solution open/close, rename and etc.
-        // Even if primary workspace is not updated, OOP will work as expected. updating primary workspace 
-        // on OOP should let latest data to be synched pre-emptively rather than on demand.
+        // Update primary workspace on OOP twice a second if VS is not running any global operation (such as build,
+        // solution open/close, rename, etc.)
         //
-        // 2 second is our usual long running interactive operation delay and 
-        // 5 second is usual ambient long running operation delay. 
-        // I chose one in between. among 3 and 4 seconds, I chose slower one - 4 second. 
+        // Even if primary workspace is not updated, other OOP queries will work as expected. Updating primary workspace
+        // on OOP should let latest data to be synced pre-emptively rather than on demand, and will kick off
+        // incremental analyzer tasks.
         //
-        // When primary workspace is staled, missing data will be synced to OOP on
-        // demand and cached for 3 min. enough for primary workspace in OOP to be synced to latest.
+        // When primary workspace is staled, missing data will be synced to OOP on demand and cached for 3 min. enough
+        // for primary workspace in OOP to be synced to latest.
         public static readonly Option<int> SolutionChecksumMonitorBackOffTimeSpanInMS = new Option<int>(
-            nameof(InternalFeatureOnOffOptions), nameof(SolutionChecksumMonitorBackOffTimeSpanInMS), defaultValue: 4000,
+            nameof(InternalFeatureOnOffOptions), nameof(SolutionChecksumMonitorBackOffTimeSpanInMS), defaultValue: 500,
```

Comment:
Is it possible that this regresses other experiences? For example, now that we much more aggressively push new snapshots to OOP, any other OOP requests, say active file diagnostics request from a subsequent edit, get delayed, taxing the whole system? As in, instead of making OOP computations and sync happen more often, it is better to work towards a pull model where any analysis/compute that needs to be responsive should have an inproc component (either in Roslyn or above the stack) that pulls on this data?

Reply:
i used this for a couple of hours today and didn't feel like anything was regressed.  

> For example, now that we much more aggressively push new snapshots to OOP, any other OOP requests, say active file diagnostics request from a subsequent edit, get delayed, taxing the whole system? 

I technically don't think this is an issue.  here's why:

First, syncing this stuff is not expensive.  i.e. we're literally saying: if anything changed, just push that over in the next half second.  It's extremely difficult for that to be a lot of work as effectively we only have one changes were possible in that time frame.

Second, when requests are actually explicitly made for some snapshot, that syncing has to happen anyways.  And syncing is complimentary.  i.e. when the 'active file diagnostics' request goes in and asks to sync the snapshot, it benefits from the preexisting pushed information that we already started on.  even if the particular file it is asking for has changed, it still gets all the other changes pushed over, and only really needs to potentially just sync over it's very latest text change.  

> it is better to work towards a pull model where any analysis/compute that needs to be responsive should have an inproc component (either in Roslyn or above the stack) that pulls on this data?